pkg/vere/main.c

```diff
@@ -1216,28 +1231,75 @@ _cw_eval(c3_i argc, c3_c* argv[])
     }
   }
 
-  printf("eval:\n");
+  fprintf(stderr, "eval:\n");
 
-  //  +wish for an eval gate (virtualized twice for pretty-printing)
-  //
-  u3_noun gat = u3v_wish("|=(a=@t (sell (slap !>(+>.$) (rain /eval a))))");
-  u3_noun res;
-  {
+  if ( c3n == jam_l ) {
+
+    //  +wish for an eval gate (virtualized twice for pretty-printing)
+    //
+    u3_noun gat = u3v_wish("|=(a=@t (sell (slap !>(+>.$) (rain /eval a))))");
+    u3_noun res;
+    {
+      u3_noun sam = u3i_string(evl_c);
+      u3_noun cor = u3nc(u3k(u3h(gat)), u3nc(sam, u3k(u3t(u3t(gat)))));
+      res = u3m_soft(0, u3n_kick_on, cor);
+    }
+  
+  
+    if ( 0 == u3h(res) ) {  //  successful execution, print output
+      u3_pier_tank(0, 0, u3k(u3t(res)));
+    }
+    else {                  //  error, print stack trace
+       u3_pier_punt_goof("eval", u3k(res));
+    }
+
+    u3z(res);
+    u3z(gat);
+  } else {
     u3_noun sam = u3i_string(evl_c);
-    u3_noun cor = u3nc(u3k(u3h(gat)), u3nc(sam, u3k(u3t(u3t(gat)))));
-    res = u3m_soft(0, u3n_kick_on, cor);
-  }
+    u3_noun res = u3v_wish_n(sam);
 
+    c3_d bits = 0;
+    c3_d len_d = 0;
+    c3_y* byt_y;
 
-  if ( 0 == u3h(res) ) {  //  successful execution, print output
-     u3_pier_tank(0, 0, u3k(u3t(res)));
-  }
-  else {                  //  error, print stack trace
-     u3_pier_punt_goof("eval", u3k(res));
-  }
+    bits = u3s_jam_xeno(res, &len_d, &byt_y);
 
-  u3z(res);
-  u3z(gat);
+    if ( c3n == khan_l ) {
+      fprintf(stderr,"jammed noun: ");
+       
+      int p=0;
```

Comment:
declare p as `size_t` since we're iterating. Refactor the while into a for-loop

Reply:
Done